cdci_data_analysis/configurer.py

```diff
@@ -33,89 +33,97 @@
 
 logger = logging.getLogger("conf")
 
-
-# TODO: this looks rather specific to INTEGRAL?
-
 class DataServerConf:
 
-    def __init__(self, data_server_url,
-                 data_server_remote_cache=None,
-                 dispatcher_mnt_point=None,
-                 dummy_cache=None,
-                 products_url=None,
-                 data_server_port=None):
-
-        if data_server_port is not None:
-            logger.warning(
-                "data_server_port is disregarded, since it is naturally included in the url")
-
-        logger.info("building config from %s %s %s", data_server_url,
-                    data_server_remote_cache, dispatcher_mnt_point)
+    def __init__(self, required_keys=None, allowed_keys=None, **kwargs):
+        if required_keys is None:
+            #temporary hardcode to preserve interface
+            required_keys = ['data_server_url', 'dummy_cache']
 
-        if data_server_url is None:
-            logger.warning(
-                f"problem constructing {self}: data_server_url is None: {data_server_url}")
+        if allowed_keys is None:
+            #temporary hardcode to preserve interface
+            allowed_optional_keys = ['optional_key']
+        else:
+            allowed_optional_keys = [x for x in allowed_keys if x not in required_keys]
 
-        if data_server_remote_cache is None:
-            logger.warning(
-                f"problem constructing {self}: data_server_remote_cache is None: {data_server_remote_cache}")
+        conf = kwargs.copy()
 
-        if dispatcher_mnt_point is None:
-            logger.warning(
-                f"problem constructing {self}: dispatcher_mnt_point is None: {dispatcher_mnt_point}")
+        logger.info("building config from %s", conf)
 
-        self.data_server_url = data_server_url
 
-        # dummy prods local cache
-        self.dummy_cache = dummy_cache
+        try:
+            self.__setattr__('data_server_url', conf.pop('data_server_url'))
```

Comment:
just out of curiosity, why __setattr__ here?

Reply:
Copy-pasted) I will change this to an assignment for clarity